### PR TITLE
Add resize scale enum variant

### PR DESCRIPTION
**Pull request by TadoTheMiner**
This pr adds the `Resize::Scale` option in #59, and adds it to the demo instead of the placeholder. I've noticed that in the code three times `static` was used when `const` could be, so I replaced it. As I changed the demo to include the `Resize::Scale` option, the gif must be re-generated

### CODE REVIEW

**Comment by benjajaja** [comment]
First of all, thank you! I really had `const` and `static` mixed up, now I see that.

You can run an approximation of the CI with `cargo make ci` if you install cargo-make. Or just run clippy.

image

This is suprising to me, I would expect the image to scale down as well instead of being cropped. I also suspect it's not really getting cropped, just overdrawn, which might cause subtle problems down the line.

Could you look into this? I think for `Scale(_)`, the "desired" area is not just the [`max(image, container)`](https://github.com/benjajaja/ratatui-image/pull/61/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R348) width and height, but instead the `image` scaled to always fit `container` and keeping the aspect ratio.

File: examples/demo/main.rs [hunk 1]
```diff
@@ -185,6 +188,25 @@ impl<'a> App<'a> {
     }
```
Comment (benjajaja) [inline examples/demo/main.rs:188]: You must add `self.image_scale_state` above here too. Cycling the images with `o` in the demo helps spot all sort of edge case bugs.

**Comment by TadoTheMiner** [comment]
By the way do we really need to have text under the images in the demo?

**Comment by benjajaja** [comment]
It scales down, but sometimes it doesn't take up the full space if you play around with window size / layout.

image

> By the way do we really need to have text under the images in the demo?

It helps a lot to eyeball glitches in different terminals. Since we're writing control sequences into some buffer, there can and have been conflicts with surrounding control sequences related to style.

**Comment by TadoTheMiner** [comment]
image
This is an issue with the library in general as it occurs in Fit aswell. So I think you should open an issue about it, that both fit and scale don't resize properly

**Comment by benjajaja** [comment]
@TadoTheMiner okay, yea there seems to be _some_ bug, although I can't exactly reproduce your screenshot.

I will merge this branch manually and push to master. I just disabled merge commits while you were working on this, sorry. I'll also add a note to the readme about this.

**Comment by benjajaja** [comment]
Rebased and merged in `main`. I'll have to check some general resize related bugs before I make a release.

**Pull request closed without merging**